src/Metrics/DefaultMetricsSender.php

```diff
@@ -31,12 +31,15 @@ public function sendMetrics(MetricsBucket $bucket): void
                 'instanceId' => $this->configuration->getInstanceId(),
                 'bucket' => $bucket->jsonSerialize(),
             ], JSON_THROW_ON_ERROR)));
+        if ($this->configuration->getProxyKey() !== null) {
+            $request = $request->withHeader('Authorization', $this->configuration->getProxyKey());
+        }
         foreach ($this->configuration->getHeaders() as $name => $value) {
             $request = $request->withHeader($name, $value);
         }
 
         try {
-            $this->httpClient->sendRequest($request);
+            $response = $this->httpClient->sendRequest($request);
```

Comment:
Why the variable assignment?

Reply:
Ah, thanks for the spot, because, I needed to see the status code, it wasn't supposed to be part of the PR, reverting :)